### PR TITLE
INTG-3224 fix api key validation

### PR DESCRIPTION
**Pull request by MickStanciu**
## Summary

Improve the way we check the validation for the API KEY

A bug was introduced a while ago with relation to api-key checking

## Checklist

- [x] Is this PR a reasonable size?

- [ ] List deployment sequence with all relevant PRs.
  - ...

---

**Code Review Guidelines** for Reviewers

- Try to review in a timely manner. Opinions/nitpicks should not be blockers. Pair on a call for non-trivial feedback.
- Overall design and approach should follow established patterns. Don't try to make the PR perfect.
- Try to identify edge cases, race conditions, over-engineering, lack of test coverage and complexity.
- If you don't feel qualified to review the code, pass it on to someone who is.

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -175,12 +175,12 @@ func (a *App) ValidateApiKey(apiKey string) string {
 
 		if err != nil {
 			runtime.LogErrorf(a.ctx, "cannot check WhoAmI: %s", err.Error())
-			return err.Error()
+			return "cannot validate the credentials for the given ApiKey"
```
Comment (HarryTranAU) [inline app.go:178]: should we be using the same error message here? This is the error to the call to WhoAmI

Comment (MickStanciu) [reply app.go:178]: I would not like to display to the user that is returned internally

**Review by HarryTranAU — APPROVED** [review]
